src/tools/fuzzing.h

```diff
@@ -2018,12 +2026,25 @@ class TranslateToFuzzReader {
         if (!wasm.functions.empty() && !oneIn(wasm.functions.size())) {
           target = pick(wasm.functions).get();
         }
-        return builder.makeRefFunc(target->name);
+        auto type = Type(HeapType(target->sig), /* nullable = */ true);
+        return builder.makeRefFunc(target->name, type);
       }
       if (type == Type::i31ref) {
         return builder.makeI31New(makeConst(Type::i32));
       }
-      return builder.makeRefNull(type);
+      if (type.isNullable()) {
+        return builder.makeRefNull(type);
+      }
+      // Well this is quite the pickle, we are asked to create a non-nullable
+      // type, and so can't just make a null. Look for an existing thing with
+      // the right type.
```

Comment:
Can we just change the fuzzer to never request a non-nullable type?

Reply:
As a reader, it can be hard to tell what is supposed to work when the code is future-facing like this. I think it would be clearer to draw a bright line and not try to anticipate our future needs.

Reply:
I guess I want to go beyond not supporting non-nullability and not even mention it in the code. Perhaps this would be beyond the scope of this PR, but it would be good to remove the concept of nullability from wasm-type.h completely, now that we've decided not to support it in the near future.

Reply:
I think we have a clear line: non-nullable types do not work, for now. And I think we know that the future needs include non-nullability, don't we?

(If somehow the wasm spec decides to remove non-nullability, then at that time of course I would agree to remove all the relevant code.)

Assuming we agree on those, I disagree that means we should delete all the nullability code from wasm-type.h, and other places with nullability code in this PR. We will need to support non-nullability pretty soon. Removing that code and adding it all back later would be a significant amount of extra work, and the only benefit seems to be to keep the code more "clean" in the short interval in between?

Maybe I'm more of a pragmatist than you. To me, the long-term goal is to support non-nullability, and I'm just trying to find the shortest path to there. Especially since "long-term" here may be weeks, I am not trying to optimize for code clarity in those interim weeks necessarily. I agree with you that that is a compromise on code clarity meanwhile, but I think we'll get to perfect clarity eventually (and I agree that that matters!). Does that make sense?

Reply:
I see. I was thinking that we might want to consider supporting non-nullability in half a year or so :P

Reply:
We resolved this in our offline discussion. We will land the forward-facing code for now, but we will go back and remove it if it looks like we won't implement nullable types for a long time.